File: tessoku/src/bin/B03.rs
use proconio::input;
use itertools::Itertools;

fn main() {
    input! {
        n: usize,
        a: [usize; n],
    }

    let res = a.iter()
        .tuple_combinations()
        .any(|(a, b, c)| a + b + c == 1000);

    println!("{}", if res { "Yes" } else { "No" });
}

File: tessoku/python/B03.py
n = int(input())
a = list(map(int, input().split()))

ok = any(a[i] + a[j] + a[k] == 1000 for i in range(n) for j in range(i + 1, n) for k in range(j + 1, n))
print("Yes" if ok else "No")
